src/TestFramework/MSTest.Core/Attributes/VSTestAttributes.cs

```diff
@@ -25,6 +23,24 @@ public enum TestTimeout
         Infinite = int.MaxValue
     }
 
+    /// <summary>
+    /// Enumeration for class initialize, that can be used with the <see cref="ClassInitializeAttribute"/> class.
+    /// Defines the behavior of the ClassInitialize methods of base classes.
+    /// The type of the enumeration must match
+    /// </summary>
+    public enum ClassInitializeInheritance
```

Comment:
I'm wondering if we should have this on Cleanup as well? Might be easy to miss this specification on initialize if they aren't near by and the naming seems to be specific for initialize although in code we implicitly stop performing cleanup too.

Reply:
@singhsarab , @jayaranigarg  : thoughts?

Reply:
Did not quite get the last part.

Reply:
well, I'm totally open to change the name to something like: `InheritanceBehavior` maybe?
I think that since initialize and cleanup are related, so their execution is/should be symmetrical and there would no need to specify the attribute in the cleanup as well, it'll do it the same fashion as initialize.

And, if we want cleanup always to be initialize (or at least every time the initialize is done) we shouldn't change this by using this. Plus, we avoid additional checks.

let me know your thoughts...

Reply:
I agree to the most part. I'm just not certain how common the case I described earlier is - Just cleanups without initialize and would users need some form of control in those cases as well? I'm inclined to have it on both. The way I see it, we hopefully would default to always use Base Class Init/Cleanup functionality soon (a major release) and not many folks would need to switch back unless its an unrecoverable break. If they do they could use as much flexibility as they can get.

Reply:
That would be great having that in a near future with the default value to `BeforeEachDerivedClass`, that way the base init and cleanup methods always execute without having to do any code change (from users perspective). On regards to the case you mentioned, given the way the current behavior is (just base class cleanup are executed only if its initialize has been executed) I don't think users would need to do anything here. Unless we change that behavior and let class cleanup method to execute even though an initialize haven't been specified.

Reply:
--
updated in latest changes.